capstone/src/main/java/com/google/sps/data/ReadData.java

```diff
@@ -1,66 +1,84 @@
-  private DatastoreService datastoreService;
   private SentimentData sentimentService;
-  private ArrayList<Entity> newEntities;
   private final String DELIMITER = ",";
 
+  /**
+   * Default constructor with no sentiment analysis dependency. Objects initialized through this
+   * constructor can be locally tested since sentiment analysis is not authorized for local testing
+   * at this time
+   */
   public ReadData() throws IOException {
-    datastoreService = DatastoreServiceFactory.getDatastoreService();
-    sentimentService = new SentimentData();
-    newEntities = new ArrayList<Entity>();
+    sentimentService = null;
+  }
+
+  /**
+   * Injected dependency included in this overloaded constuctor. Objects initialized through this
+   * constructor will use sentiment analysis to analyze positivity/ negativity of survey responses.
+   */
+  public ReadData(SentimentData sentimentService) throws IOException {
+    this.sentimentService = sentimentService;
   }
 
   /**
-   * Calls read file while iterating through all the zip codes
+   * Calls read file while iterating through all files in the assets directory
    *
-   * @return {Void}
+   * @return ArrayList<Entity> all the entities that were created from parsing every file in the
+   *     assets directory
    */
-  public void readAll() {
-    Set<String> zipCodes = MapData.zipPrecinctMap.keySet();
+  public ArrayList<Entity> allEntitiesFromFiles() throws IOException {
+    ArrayList<Entity> allEntities = new ArrayList<Entity>();
 
-    for (String zip : zipCodes) {
-      String filePath = "assets/" + zip + ".csv";
+    File folder = new File("assets/");
 
-      try {
-        readFile(new FileReader(filePath), zip);
-      } catch (FileNotFoundException e) {
-        System.out.println(filePath + " is not in the assets folder");
-      }
+    File[] fileNames = folder.listFiles();
+    for (File file : fileNames) {
+      entitiesFromFile(allEntities, file);
     }
 
-    sentimentService.close();
+    if (sentimentService != null) {
+      sentimentService.close();
+    }
+
+    return allEntities;
   }
 
   /**
-   * Reads the survey csv, splits the lines by a comma delimiter, and adds each element to data
-   * store as an entity property
+   * Reads the survey csv, splits the lines by a comma delimiter, and creates entites with the
+   * parsed data for datastore
    *
    * @param file This is the file that is being parsed
-   * @param zip This is the zip code the file is associated with
-   * @return {Void}
+   * @param newEntities stores created entities
+   * @return void
    */
-  public void readFile(FileReader file, String zipCode) {
-    final int EXPECTED_LENGTH = 15;
+  public void entitiesFromFile(ArrayList<Entity> newEntities, File file) {
+    String zipCode = file.getName().substring(0, 5);
+    FileReader fileReader = null;
+
+    try {
+      fileReader = new FileReader(file);
+    } catch (FileNotFoundException e) {
+      System.out.println("No file found");
```

Comment:
Should we just return early here if file isn't found or raise the exception?

Reply:
I did not think of that, thank you for the suggestion!